### ISSUE TITLE
Training scripts for baseline systems

### ISSUE THREAD

**Issue by ajk1402** [piotrkawa/audio-deepfake-source-tracing#2]
Hi 

I have following questions as given below,

1. is train_ge2e.py works with data augmentation ?, where is the eval script for model trained with ge2e?
2. what are the suggestions from your side to make end2end working for source tracing task with input as wav, in training loop from data augmentation point of view?

Regards
Ajinkya Kulkarni

**Comment by piotrkawa** [comment]
Hi,
> is train_ge2e.py works with data augmentation ?, where is the eval script for model trained with ge2e?

The existing `train_ge2e.py` does not include the data augmentation pipeline - it focuses on training feature extractor that can be used in next steps like clustering or multi-class classification. The script reports loss on the test subset of the dataset.

> what are the suggestions from your side to make end2end working for source tracing task with input as wav, in training loop from data augmentation point of view?

Using the provided code there are two general approaches. Both of them use wavs as input.

1. Training a feature extractor (`train_ge2e.py`) and then supporting some operations on the obtained embeddings:
     1. to classify embeddings using neural networks or algorithms like k-nearest neighbors,
     2. for clustering task.

2. Using the REFD approach (`train_refd.py`) - it supports e2e multi-class classification and includes an out-of-domain detector to identify samples likely outside of trained classes.

Best regards,
Piotr Kawa

**Comment by ajk1402** [comment]
Thanks for the response.

Ajinkya
